Microsoft.Toolkit.Uwp/IncrementalLoadingCollection/IncrementalLoadingCollection.cs

```diff
@@ -196,31 +196,41 @@ public IAsyncOperation<LoadMoreItemsResult> LoadMoreItemsAsync(uint count)
             => LoadMoreItemsAsync(count, new CancellationToken(false)).AsAsyncOperation();
 
         /// <summary>
-        /// Clears the collection and reloads data from the source
+        /// Clears the collection and triggers/forces a reload of the first page
         /// </summary>
         /// <returns>This method does not return a result</returns>
-        [Obsolete("RefreshAsync is deprecated, please use Refresh instead.")]
-        public async Task RefreshAsync()
-        {
-            await Task.Run(() => Refresh());
-        }
-
-        /// <summary>
-        /// Clears the collection and resets the page index
-        /// which triggers an automatic reload of the first page
-        /// </summary>
-        public void Refresh()
+        public Task RefreshAsync()
         {
             if (IsLoading)
             {
                 _refreshOnLoad = true;
             }
             else
             {
+                var previousCount = Count;
                 Clear();
                 CurrentPageIndex = 0;
                 HasMoreItems = true;
+
+                if (previousCount == 0)
+                {
+                    // When the list was empty before clearing, the automatic reload isn't fired, so force a reload.
+                    var loadMoreItemsTask = LoadMoreItemsAsync(0).AsTask();
```

Comment:
Just return LoadMoreItemsAsync
`return LoadMoreItemsAsync(0);`

Reply:
I can't call the ConfigureAwait in the following line if it's not a task, and removing that call causes a deadlock when calling the Refresh method from the UI thread.

Reply:
Sorry, you're right. Previously I was waiting the task on the Refresh method and that was causing the actual deadlock, not the context. Anyhow, I still need to return it as a task as it doesn't have a return value (required for IAsyncOperation), and I'd like to avoid copying the same call used inside that method.